### PR TITLE
Modify replace-im-links to swap re-named links too

### PR DESCRIPTION
**Pull request by mjoss17**

### CODE REVIEW

**Comment by mjoss17** [comment]
I added a test case to the end of this pr to test that that media.json hashes are created in the same way that these patching hashes are created. To do so I ran through all three commands to `fetch-im-resources`, `copy-resources-s3`, and 'replace-im-links' in full, since the media.json hashes are filled during 'copy-resources-s3' using the directory outputted by 'fetch-resources-s3'. Its rather long but I thought it could be a helpful run through (maybe).

If this is too ugly I think we can check whether the hashes are created the same by just calling the helper - `copy_resources_s3.new_resource_hashes()` which is what 'copy-reosources-s3' uses internally, and comparing its output to the new patch.

File: tests/test_replace_im_links.py [hunk 1]
```diff
@@ -373,3 +376,157 @@ def test_fetch_im_resource_extracted_no_matches(
...
+            data = file.read()
+            for elem in html.fragments_fromstring(
+                        data)[0].xpath('//*[@src]'):
+                resources.append(elem.attrib['src'])
+    assert len(resources) == 2
+    for r in resources:
+        assert new_prefix in r
+
+
+def test_fetch_and_replace_im_resources_html_overlapping_filenames(
```
Comment (rnathuji) [inline tests/test_replace_im_links.py:438]: @mjoss17 - Ah, I see now what you meant by the test case you added. It actually doesn't seem too bad to me in terms of maintainability versus the value it provides in terms of providing some sanity checking that the three related commands "fit" together. Thank you for thinking of / adding this!

File: mbtools/replace_im_links.py [hunk 2]
```diff
@@ -7,6 +9,24 @@
...
+    for elem in utils.find_elements_containing(content_tree, src_content):
+        im_filename = elem.attrib["src"]
+        if im_filename in swap_mapping.keys():
+            num_changes += 1
+            elem.attrib["src"] = swap_mapping[im_filename]
+        else:
+            # Fetch link, calculate sha1, and replace link anyway.
+            data = requests.get(im_filename).content
+            sha1 = hashlib.sha1(data).hexdigest()
+            for key in swap_mapping.keys():
```
Comment (rnathuji) [inline mbtools/replace_im_links.py:23]: I didn't immediately realize this, but it looks like you're iterating here making this lookup `O(n)`. It also makes an implicit assumption that the target URL includes the sha1 value. Would it be worthwhile to perhaps have `parse_media_file` return two mapping objects?:

1. The existing IM URL -> `entry['s3_url']`
2. A new `entry['sha1']` -> `entry['s3_url']`

You could then pass both mappings into this function and make this branch a bit more efficient and perhaps improve clarity :man_shrugging: .

File: tests/test_replace_im_links.py [hunk 3]
```diff
@@ -1,5 +1,8 @@
+import botocore.stub
+import boto3
 import os
-from mbtools import replace_im_links
+from mbtools import copy_resources_s3, replace_im_links
+from mbtools import fetch_im_resources
```
Comment (rnathuji) [inline tests/test_replace_im_links.py:5]: Minor: You can combine these to be `from mbtools import copy_resources_s3, replace_im_links, fetch_im_resources`

**Review by rnathuji — APPROVED** [review]

**Pull request merged**